src/v2/view-builder/views/shared/polling.js

```diff
@@ -3,28 +3,30 @@ import { _ } from 'okta';
 export default {
   startPolling () {
     // Factor and Authenticator won't co-exists hence it's safe to trigger both.
-    this._startFactorPolling();
     this._startAuthenticatorPolling();
   },
 
-  _startFactorPolling () {
-    const factor = this.options.appState.get('factor');
-    const factorPollingInterval = factor && factor.poll && factor.poll.refresh;
-    if (_.isNumber(factorPollingInterval)) {
-      this.polling = setInterval(()=>{
-        this.options.appState.trigger('invokeAction', 'factor-poll');
-      }, factorPollingInterval);
-    }
-  },
-
   _startAuthenticatorPolling () {
-    const authenticator = this.options.appState.get('authenticator');
-    const pollInterval = authenticator && authenticator.poll && authenticator.poll.refresh;
-    if (_.isNumber(pollInterval)) {
-      this.polling = setInterval(()=>{
-        this.options.appState.trigger('invokeAction', 'authenticator-poll');
-      }, pollInterval);
-    }
+    // Factor and Authenticator won't co-exists hence it's safe to trigger both.
+    [
+      'currentAuthenticator',
+      'currentAuthenticatorEnrollment',
+      'factor',
+    ].some(responseKey => {
+      if (this.options.appState.has(responseKey)) {
```

Comment:
nit: Instead of if-else, this makes the code look better. But that's just IMO
if (!this.options.appState.has(responseKey)) {
return false;
}
// Rest of truthy code..